fully-client-side-calculator/index.js

```diff
@@ -0,0 +1,141 @@
+const screen = document.getElementById("screen");
+const numbers = document.querySelectorAll(".numbers-to-press");
+const operandButtons = document.querySelectorAll(".operation-buttons");
+const equalsButton = document.getElementById("equalsButton");
+const clearButton = document.getElementById("clearButton");
+const decimalButton = document.getElementById("decimalButton");
+let secondValue = "";
+let operand = "";
+let currentValue = 0;
+let operandPressed = false;
+let timesOperandUsed = 0;
+let equalsButtonPurposelyPressed = false;
+let nothingHappened = true;
+const calculator = new Calculator();
+
+function returnValueOfOperation() {
+  if (screen.value === "") {
+    return;
+  }
+  currentValue = parseFloat(secondValue);
+
+  switch (operand) {
+    case "+":
+      calculator.add(currentValue);
+      break;
+    case "-":
+      calculator.subtract(currentValue);
+      break;
+    case "*":
+      calculator.multiply(currentValue);
+      break;
+    case "/":
+      calculator.divide(currentValue);
+      break;
+    case "^":
+      calculator.power(currentValue);
+      break;
+  }
+
+  screen.value = calculator.getTotal();
+}
+
+clearButton.addEventListener("click", () => {
+  screen.value = "0";
```

Comment:
It would be good to have a clear function defined that did this. The clear button should call the clear function and then also call calculator.clear() separately.

Then you would call the clear function above instead of initialising these variables manually. You would just do `let blah, blah2, blah3;`

Reply:
I have created the clear function as you have instructed, however I don't understand what you mean by Initialising these variables manually by doing let blah, blah2, blah3? If I did that then would the rest of the variables not be accessible to the rest of the functions that are being used due to scope rules?